Add config flow to Wake on LAN (#33660)
home-assistant/home-assistant.github.io: source/_integrations/wake_on_lan.markdown

- [Switch](#switch)

{% include integrations/config_flow.md %}

{% configuration_basic %}
Mac address:

>   description: "The MAC address to send the wake up command to, e.g., `00:01:02:03:04:05`."

Reviewer: ```suggestion
  description: "The MAC address to send the wake-up command to. For example, `00:01:02:03:04:05`."
```

host:
  description: The IP address or hostname to check the state of the device (on/off). If this is not provided, the state of the switch will be assumed based on the last action that was taken.
turn_off:
  description: Defines an [action](/getting-started/automation/) to run when the switch is turned off.
broadcast_address:
  description: The IP address of the host to send the magic packet to.
broadcast_port:
  description: The port to send the magic packet to.

> {% configuration_basic %}

Author: ```suggestion
{% endconfiguration_basic %}
```

> {% configuration_basic %}
> mac:
>   description: "The MAC address to send the wake up command to, e.g, `00:01:02:03:04:05`."
> host:
>   description: The IP address or hostname to check the state of the device (on/off). If this is not provided, the state of the switch will be assumed based on the last action that was taken.
> turn_off:
>   description: Defines an [action](/getting-started/automation/) to run when the switch is turned off.
> broadcast_address:
>   description: The IP address of the host to send the magic packet to.
> broadcast_port:
>   description: The port to send the magic packet to.
> {% endconfiguration_basic %}

Author: ```suggestion
{% configuration_basic %}
Mac address:
  description: "The MAC address to send the wake up command to, e.g., `00:01:02:03:04:05`."
Host:
  description: The IP address or hostname to check the state of the device (on/off). If this is not provided, the state of the switch will be assumed based on the last action that was taken.
Turn off action:
  description: Defines an [action](/getting-started/automation/) to run when the switch is turned off.
Broadcast address:
  description: The IP address of the host to send the magic packet to.
Broadcast port:
  description: The port to send the magic packet to.
{% endconfiguration_basic %}
```

The service to route the packet is most likely named "IP Helper". It may support Wake on LAN, but not all routers support this.
{% endnote %}

## Button

The `wake_on_lan` (WOL) switch {% term integration %} allows you to turn on a [WOL](https://en.wikipedia.org/wiki/Wake-on-LAN) enabled computer.

The WOL button can only turn on your computer.

> It will send a magic packet to the mac-address specified in the configuration and as a button it is state-less meaning it can not monitor if the WOL enabled computer has actually received the wake-up call and has started.

Reviewer: ```suggestion
It will send a magic packet to the MAC address specified in the configuration. As a button, it is stateless. This means it can not monitor if the WOL-enabled computer has actually received the wake-up call and has started.
```
